### PR TITLE
Code refactor in all projects

### PR DESCRIPTION
**Pull request by GermanAizek**
This PR will include removing unused code, refining and correcting some parts, and optimizing the code.

Signed-off-by: GermanAizek <GermanAizek@yandex.ru>

### CODE REVIEW

**Comment by pent0** [comment]
Tell me if it's completed, since I don't see any unused code removals yet. Try to squash commits with same purpose together.

**Comment by GermanAizek** [comment]
@pent0,
I'm not done yet, I'll let you know when commits will be enough.

**Comment by GermanAizek** [comment]
@pent0,
What is this Russian proposal for? I don't see a connection to the nearest code.

Translation into English:

These are insults, the first word insults a person, and the second is said to increase emotions.

https://github.com/EKA2L1/EKA2L1/blob/ec4a25be13574a29d8a3e896463156bfb2c5ec61/src/emu/kernel/include/kernel/process.h#L181
https://github.com/EKA2L1/EKA2L1/blob/ec4a25be13574a29d8a3e896463156bfb2c5ec61/src/emu/kernel/include/kernel/process.h#L182

**Comment by GermanAizek** [comment]
Delete or comment func?

https://github.com/EKA2L1/EKA2L1/blob/d65c2005febcb34789f046c9b681db28ec551474/src/emu/package/src/sis_script_interpreter.cpp#L54-L56

**Review by GermanAizek — COMMENTED** [review]
@pent0, check it.

File: src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace eka2l1::drivers {
         glGetIntegerv(GL_MAJOR_VERSION, &major_gl);
 
         if (!is_gles && (major_gl < 4)) {
-            std::vector<std::string> GL_REQUIRED_EXTENSIONS = {
+            std::list<std::string> GL_REQUIRED_EXTENSIONS = {
```
Comment (pent0) [inline src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp:51]: This is wasting allocations as a double-linked list. No reason for this

File: src/emu/kernel/include/kernel/process.h [hunk 2]
```diff
@@ -168,7 +168,7 @@ namespace eka2l1::kernel {
         common::roundabout thread_list;
         common::roundabout codeseg_list;
 
-        std::vector<kernel::process*> child_processes_;
+        std::list<kernel::process*> child_processes_;
```
Comment (pent0) [inline src/emu/kernel/include/kernel/process.h:171]: This one is okay

**Comment by pent0** [comment]
Answer questions:
> What is this Russian proposal for? I don't see a connection to the nearest code.

I put random things related to the event occurred during that day, somehow that sentence I don't remember the context. Delete is your choice, I won't mind.

> Delete or comment func?
If unused you can delete

You can contact me on Discord in case I don't respond here (I am not available on this Github account most of the time nowadays)

**Comment by pent0** [comment]
Other than that it's fine. I try to contact you but to note that you are not available on the server anymore. Very sorry for my late respond.

**Comment by GermanAizek** [comment]
@pent0, pls do not close PR I am still busy, I will let you know how I will be free.

File: src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp (hunk 1 above)
Comment (GermanAizek) [reply src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp]: @pent0, fix it https://github.com/EKA2L1/EKA2L1/pull/303/commits/57ead7f3759945dfddd1235ed073901617ae7d46

File: src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp [hunk 3]
```diff
@@ -48,7 +48,7 @@ namespace eka2l1::drivers {
         glGetIntegerv(GL_MAJOR_VERSION, &major_gl);
 
         if (!is_gles && (major_gl < 4)) {
-            std::list<std::string> GL_REQUIRED_EXTENSIONS = {
+            std::vector<std::string> GL_REQUIRED_EXTENSIONS = {
```
Comment (pent0) [inline src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp]: Better to be an array like you did before

File: src/emu/services/src/audio/keysound/keysound.cpp [hunk 4]
```diff
@@ -345,7 +345,7 @@ namespace eka2l1 {
     }
 
     void keysound_session::lock_context(eka2l1::service::ipc_context *ctx) {
-        ctx->complete(epoc::error_none);
```
Comment (pent0) [inline src/emu/services/src/audio/keysound/keysound.cpp:348]: This just break its meaning. Please just let code be stubbed as it is

File: src/emu/cpu/src/arm_dynarmic.cpp [hunk 5]
```diff
@@ -74,7 +74,7 @@ namespace eka2l1::arm {
         }
 
         CallbackOrAccessTwoWords CompileGetTwoWords(bool two, unsigned opc, coproc_reg CRm) override {
-            return CallbackOrAccessTwoWords{};
+            return CompileSendTwoWords(two, opc, CRm);
```
Comment (pent0) [inline src/emu/cpu/src/arm_dynarmic.cpp:77]: This too. Don't change meaning where not needed.

File: src/emu/services/src/audio/keysound/keysound.cpp (hunk 4 above)
Comment (GermanAizek) [reply src/emu/services/src/audio/keysound/keysound.cpp]: @pent0 check it https://github.com/EKA2L1/EKA2L1/commit/e69502495791ac4f7c99fa21d58ad16deb6e435e

File: src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp (hunk 3 above)
Comment (GermanAizek) [reply src/emu/drivers/src/graphics/backend/ogl/graphics_ogl.cpp]: @pent0 fix it https://github.com/EKA2L1/EKA2L1/commit/e69502495791ac4f7c99fa21d58ad16deb6e435e

File: src/emu/cpu/src/arm_dynarmic.cpp (hunk 5 above)
Comment (GermanAizek) [reply src/emu/cpu/src/arm_dynarmic.cpp]: @pent0 fix it https://github.com/EKA2L1/EKA2L1/commit/e69502495791ac4f7c99fa21d58ad16deb6e435e

**Comment by pent0** [comment]
I suggest rebase, the project is being frozen at the moment with new code, ImGui is already removed :)

**Comment by pent0** [comment]
You can reopen again when you have interest. I'm sorry((( I will take some assumption here.

**Comment by GermanAizek** [comment]
@pent0, OK, I'll recreate PR from active branch later.

**Pull request closed without merging**